packages/circuit-ui/components/Button/Button.tsx

```diff
@@ -287,12 +382,26 @@ export const Button = forwardRef(
     return (
       <StyledButton
         {...props}
+        {...(loadingLabel && {
+          'disabled': isLoading,
+          'aria-live': 'polite',
+          'aria-busy': isLoading,
+        })}
         ref={ref}
         as={props.href ? Link : 'button'}
         onClick={handleClick}
       >
         {Icon && <Icon css={iconStyles} role="presentation" />}
-        {children}
+        {loadingLabel ? (
+          <Fragment>
+            <LoadingIcon isLoading={Boolean(isLoading)} size="byte">
+              <LoadingLabel>{loadingLabel}</LoadingLabel>
+            </LoadingIcon>
+            <Children isLoading={Boolean(isLoading)}>{children}</Children>
+          </Fragment>
+        ) : (
+          children
+        )}
```

Comment:
The whole button content should be hidden when the button is in the loading state, including the `icon`. I'd always wrap the content in a wrapper (`Children` in this case, although I would rename it e.g. to `Content`) to keep the styles consistent.